content/posts/pathlib.md

```diff
@@ -0,0 +1,201 @@
++++ 
+draft = false
+date = 2021-09-01T23:53:53-04:00
+title = "The Advantages of Pathlib"
+description = "Using Python's Pathlib library"
+slug = ""
+authors = ["Tom"]
+tags = ["Python","Pathlib","os.path"]
+categories = []
+externalLink = ""
+series = ["python"]
++++
+
+#### Python3 has a standard library with classes for filepaths. Are you using it yet?
+
+If you have some experience using Python, you probably already know it has some good tools for ironing out differences between Windows and Unix paths, provided you don't build paths like this:
+
+{{< highlight python >}}
+path = basepath + '/never' + '/do' + '/this'
+{{</ highlight >}}
+
+The traditional answer has been to use libraries like [os](https://docs.python.org/3/library/os.html) and [os.path](https://docs.python.org/3/library/os.path.html):
+
+{{< highlight python >}}
+from os.path import join
+path = join(basepath, 'a', 'better', 'path')
+{{</ highlight >}}
+
+But in terms of ease of use, these libraries are starting to show their age.
+Python3 includes [pathlib](https://docs.python.org/3/library/pathlib.html), a more convenient class-based library for interacting with paths.
+
+##### Pathlib Classes
+
+By importing pathlib and using a Path class, we'll get a *concrete* class based on the underlying filesystem.
```

Comment:
@AnUnpronounceableName is this an example of the "pathlib" font inconsistency you were mentioning?

Reply:
This is one example, yeah